.github/workflows/dotnet-upgrade.yml

```diff
@@ -0,0 +1,167 @@
+      SHOULD_UPDATE: ${{ steps.fetch_latest_patch.outputs.SHOULD_UPDATE }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Check patch version
+        id: fetch_latest_patch
+        shell: bash
+        run: |
+          latest_patch_version=$(curl -sb -H "Accept: application/json" "https://dotnetcli.blob.core.windows.net/dotnet/Sdk/${{ env.MAJOR_MINOR_VERSION }}/latest.version")
+          current_patch_version=$(jq .sdk.version ./src/global.json | xargs)
+
+          exit_code=0
+          function version { printf "%03d%03d%03d%03d" $(echo "$1" | tr '.' ' '); }
+          
+          [ $(version $current_patch_version) -ge $(version $latest_patch_version) ] && exit_code=1
+
+          echo "DOTNET_CURRENT_VERSION=${current_patch_version}" >> $GITHUB_OUTPUT
+          echo "DOTNET_LATEST_VERSION=${latest_patch_version}" >> $GITHUB_OUTPUT
+          echo "SHOULD_UPDATE=${exit_code}" >> $GITHUB_OUTPUT
+      - name: Create annotation for build error
+        if: ${{ steps.fetch_latest_patch.outputs.SHOULD_UPDATE == 1 }}
+        run: echo "::warning ::Latest DotNet SDK patch is ${{ steps.fetch_latest_patch.outputs.DOTNET_LATEST_VERSION }}, and we are on ${{ steps.fetch_latest_patch.outputs.DOTNET_CURRENT_VERSION }}. No need to update"
+      - name: Update patch version
+        if: ${{ steps.fetch_latest_patch.outputs.SHOULD_UPDATE == 0 }}
+        shell: bash
+        run: |
+          patch_version="${{ steps.fetch_latest_patch.outputs.DOTNET_LATEST_VERSION }}"
+          current_version="${{ steps.fetch_latest_patch.outputs.DOTNET_CURRENT_VERSION }}"
+
+          # Update globals
+          echo $(cat ./src/global.json | jq --arg patch_version "$patch_version" '.sdk.version = $patch_version') > ./src/global.json
+
+          # Update devcontainer
+          echo $(cat ./.devcontainer/devcontainer.json | jq --arg patch_version "$patch_version" '.features."ghcr.io/devcontainers/features/dotnet".version = $patch_version') > ./.devcontainer/devcontainer.json
+
+          # Update dev.sh
+          sed -i "s/DOTNETSDK_VERSION=\"$current_version\"/DOTNETSDK_VERSION=\"$patch_version\"/g" ./src/dev.sh
+      - name: GIT commit and push all changed files
+        if: ${{ steps.fetch_latest_patch.outputs.SHOULD_UPDATE == 0 }}
+        id: create_branch
+        env: 
+          CI_COMMIT_MESSAGE: Upgrade dotnet sdk to v${{ steps.fetch_latest_patch.outputs.DOTNET_LATEST_VERSION }}
+          CI_COMMIT_AUTHOR: AutoUpdater
+          BRANCH_NAME: feature/dotnetsdk-upgrade/${{ steps.fetch_latest_patch.outputs.DOTNET_LATEST_VERSION }}
+        run: |
+          git config --global user.name "${{ env.CI_COMMIT_AUTHOR }}"
+          git config --global user.email "username@users.noreply.github.com"
+
+          git checkout -b ${{ env.BRANCH_NAME }}
+          git commit -a -m "${{ env.CI_COMMIT_MESSAGE }}"
+          git push --set-upstream origin ${{ env.BRANCH_NAME }}
```

Comment:
do we need to force push in case we already have a branch created?
or not pushing the branch in case a PR that bump the dotnet sdk version already exists.

Reply:
I think I'll fail out if the branch already exists, just in case someone is working on it for whatever reason